src/main.rs

```diff
@@ -3964,4 +3998,127 @@ mod tests {
+    }
+    #[test]
+    fn type_directive_on_property() {
+        let src = "
+            export type TeamId = number;
+            export interface A { 
+                // @typescript-to-gdscript-type: unknown
+                interface_value: TeamId; 
+            }
+        ";
+        let models = parse_and_get_models("type-directive-on-property.ts", &src);
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("unknown")));
+    }
+    #[test]
+    #[ignore]
+    fn type_directive_override_property() {
+        let src = "
+            // @typescript-to-gdscript-type: int
+            export type TeamId = number;
+            export interface A { 
+                // @typescript-to-gdscript-type: unknown
+                interface_value: TeamId; 
+            }
+        ";
+        let models = parse_and_get_models("type-directive-override-property.ts", &src);
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("unknown")));
+    }
+
+    #[test]
+    fn type_directive_on_property_imported() {
+        let team_id = "
+            export type TeamId = number;
+        ";
+        let src = "
+            import { TeamId } from \"./team-id.ts\";
+            export interface A {
+                // @typescript-to-gdscript-type: unknown
+                interface_value: TeamId;
+            }
+        ";
+        let models = parse_and_get_models_with_imports(
+            "type-directive-on-property-imported.ts",
+            &src,
+            &[&parse_from_string("team-id.ts", team_id)],
+        );
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("unknown")));
+    }
+
+    #[test]
+    fn type_directive_override_property_imported() {
+        let team_id = "
+            // @typescript-to-gdscript-type: int
+            export type TeamId = number;
+        ";
+        let src = "
+            import { TeamId } from \"./team-id.ts\";
+            export interface A {
+                // @typescript-to-gdscript-type: unknown
+                interface_value: TeamId;
+            }
+        ";
+        let models = parse_and_get_models_with_imports(
+            "type-directive-override-property-imported.ts",
+            &src,
+            &[&parse_from_string("team-id.ts", team_id)],
+        );
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("unknown")));
+    }
+
+    #[test]
+    #[ignore]
+    fn type_directive_on_builtin_property() {
```

Comment:
This test doesn't pass on main either, so I just added an ignored test. We probably aren't going to encounter this much anyways.